src/lib/components/Swap/SwapPropValidator.tsx

```diff
@@ -0,0 +1,66 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { DefaultAddress, SwapProps } from 'lib/components/Swap'
+import { IntegrationError } from 'lib/errors'
+import { PropsWithChildren, useEffect } from 'react'
+
+import { isAddress } from '../../../utils'
+
+function isAddressOrAddressMap(addressOrMap: DefaultAddress): boolean {
+  if (typeof addressOrMap === 'object') {
+    return Object.values(addressOrMap).every((address) => isAddress(address))
+  }
+  if (typeof addressOrMap === 'string') {
+    return typeof isAddress(addressOrMap) === 'string'
+  }
+  return false
+}
+
+type ValidatorProps = PropsWithChildren<SwapProps>
+
+export default function SwapPropValidator(props: ValidatorProps) {
+  // convenience fee constraints
+  const { convenienceFee, convenienceFeeRecipient } = props
+  useEffect(() => {
+    if (convenienceFee) {
+      if (convenienceFee > 100 || convenienceFee < 0) {
+        throw new IntegrationError('convenienceFee must be between 0 and 100')
```

Comment:
nit: append `(was ${convenienceFee}).` (don't forget the period for consistency)
do this in all IntegrationErrors for this validator.